csrc/include/speculative_sampling.cuh

```diff
@@ -0,0 +1,168 @@
+// SPDX-License-Identifier: MIT
```

Comment:
This file comes from sglang (and the chain speculative sampling function in flashinfer) and you should keep the source license like:
https://github.com/sgl-project/sglang/blob/main/sgl-kernel/csrc/speculative/speculative_sampling.cuh#L1-L16